### PR TITLE
Map changes

### PR DESCRIPTION
**Pull request by oksanana**
Fixes #

### Changes

Summary of changes you made.

* Change 1
* Change 2

### Test plan

Explain how to test changes you made.

* [ ] Step 1
* [ ] Step 2

### Demo (optional)

Sreenshots (before & after) or video that show result of your work.

### CODE REVIEW

File: src/client/components/Maps/Map.js [hunk 1]
```diff
@@ -15,6 +14,7 @@ import { getIsMapModalOpen, getSuitableLanguage } from '../../reducers';
 import { setMapFullscreenMode } from './mapActions';
 import './Map.less';
 import mapProvider from '../../helpers/mapProvider';
+import CustomMarker from './CustomMarker';
```
Comment (milklim) [inline src/client/components/Maps/Map.js]: import styles at the last line of imports

File: src/client/objectTypes/objectTypeActions.js [hunk 2]
```diff
@@ -61,6 +62,33 @@ export const getObjectType = (typeName, { limit = 15, skip = 0 } = { limit: 15,
   });
 };
 
+export const getObjectTypeMap = ({ limit = 30, skip = 0, map = {} } = { limit: 30, skip: 0, map: {} }) => (
```
Comment (milklim) [inline src/client/objectTypes/objectTypeActions.js]: duplicated code (with previous action). Need to refactor

File: src/client/components/Maps/Map.js (hunk 1 above)
Comment (oksanana) [reply src/client/components/Maps/Map.js]: changed

File: src/client/objectTypes/objectTypeActions.js (hunk 2 above)
Comment (oksanana) [reply src/client/objectTypes/objectTypeActions.js]: changed

File: src/client/components/Navigation/TopNavigation.js [hunk 3]
```diff
@@ -41,62 +42,75 @@ const TopNavigation = ({ authenticated, location: { pathname } }) => {
...
-          to={LINKS.ABOUT}
-          className={classNames('TopNavigation__link', {
-            'TopNavigation__link--active': pathname.includes(LINKS.ABOUT),
-          })}
-        >
-          <FormattedMessage id="about" defaultMessage="About" />
-        </Link>
-      </li>
-    </ul>
+          <ul className="TopNavigation__menu center topNav">
```
Comment (milklim) [inline src/client/components/Navigation/TopNavigation.js]: we can omit 'topNav' class

File: src/client/components/Maps/CustomMarker/CustomMarker.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+const CustomMarker = ({ payload, anchor, onMouseOut, onMouseOver, onClick, onContextMenu, left, top, hover, isMarked }) => {
+  const [hoverImg, setHoverImg] = useState(false);
+  const eventParameters = (event) => ({
+    event,
+    anchor,
+    payload
+  });
+
+  // controls
+  const isRetina = () => typeof window !== 'undefined' && window.devicePixelRatio >= 2;
```
Comment (milklim) [inline src/client/components/Maps/CustomMarker/CustomMarker.js]: move to useEffect() or outside the function

File: src/client/components/Maps/Map.js [hunk 5]
```diff
@@ -35,52 +37,79 @@ class MapOS extends React.Component {
...
-    !_.isEmpty(props.wobjects)
-      ? _.map(props.wobjects, wobject => {
+  setRadius = zoom => {
+    this.setState({radius: getRadius(zoom)});
+  };
+
+  setRadiusFullscreen = zoom => {
+    const { width } = this.props;
+    this.setState({ radius: getRadius(zoom) * this.mapRef.current.state.width/width});
+  };
```
Comment (milklim) [inline src/client/components/Maps/Map.js]: Combine these methods

File: src/client/components/Maps/Map.js [hunk 6]
```diff
@@ -122,10 +151,19 @@ class MapOS extends React.Component {
     this.setState({ infoboxData: null });
   };
 
-  incrementZoom = () =>
-    this.state.zoom < 20 ? this.setState({ zoom: this.state.zoom + 1 }) : null;
+  incrementZoom = () => {
+    const { isFullscreenMode } = this.props;
+    return this.state.zoom < 18
+      ? this.setState({ zoom: this.state.zoom + 1 }, () =>  (isFullscreenMode ? this.setRadiusFullscreen(this.state.zoom) : this.setRadius(this.state.zoom)))
```
Comment (milklim) [inline src/client/components/Maps/Map.js]: why do we return setState

File: src/client/components/Maps/Map.js [hunk 7]
```diff
@@ -122,10 +151,19 @@ class MapOS extends React.Component {
...
+      : null;
+  };
 
-  decrementZoom = () => (this.state.zoom > 0 ? this.setState({ zoom: this.state.zoom - 1 }) : null);
+  decrementZoom = () => {
+    const { isFullscreenMode } = this.props;
+    return this.state.zoom > 0
+      ? this.setState({ zoom: this.state.zoom - 1 }, () =>  (isFullscreenMode ? this.setRadiusFullscreen(this.state.zoom) : this.setRadius(this.state.zoom)))
```
Comment (milklim) [inline src/client/components/Maps/Map.js]: same

File: src/client/components/Navigation/TopNavigation.js [hunk 8]
```diff
@@ -41,62 +42,75 @@ const TopNavigation = ({ authenticated, location: { pathname } }) => {
...
-            'TopNavigation__link--active':
-              pathname === '/' || FEED_URLS.some(feedUrl => pathname.includes(feedUrl)),
-          })}
+    <div className="TopNavigation">
+      <div className='container menu-layout'>
+        <Scrollbars
+          style={{ width: '100%', height: 46 }}
+          universal
+          autoHide
+          renderView={({ style}) => (
```
Comment (milklim) [inline src/client/components/Navigation/TopNavigation.js]: space before bracket

File: src/client/objectTypes/objectTypeReducer.js [hunk 9]
```diff
@@ -59,6 +60,25 @@ const objectType = (state = initialState, action) => {
         fetching: false,
       };
     }
+    case wobjTypeActions.GET_OBJECT_TYPE_MAP.SUCCESS: {
+      const { locale } = action.meta;
+      const {
+        related_wobjects: relatedWobjects,
+      } = action.payload;
+      const filteredObjects = [
+        ...relatedWobjects
```
Comment (milklim) [inline src/client/objectTypes/objectTypeReducer.js]: we can omit spread operator

File: src/client/rewards/Rewards.js [hunk 10]
```diff
@@ -2,7 +2,7 @@
 import { message } from 'antd';
 import React from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import {connect} from 'react-redux';
```
Comment (milklim) [inline src/client/rewards/Rewards.js]: spaces before brackets

File: src/client/rewards/Rewards.js [hunk 11]
```diff
@@ -138,8 +142,12 @@ class Rewards extends React.Component {
     }
   }
 
+  setMapArea = mapArea => this.props.getObjectTypeMap(mapArea);
+
   getRequiredObjects = () =>
-    map(this.state.propositions, proposition => proposition.required_object);
+    this.state.propositions
+      .filter(proposition => proposition.required_object)
+      .map(proposition => ({ ...proposition.required_object, campaigns: {}})); // for rendering wobjects with campaigns
```
Comment (milklim) [inline src/client/rewards/Rewards.js]: // add 'campaigns' prop to display object on the map with proper marker

**Pull request merged**